### DISCUSSION TITLE
How can I use tidy.js with rmarkdown?

### DISCUSSION THREAD

**Discussion by pbeshai** [pbeshai/tidy#4] [Q&A] [answered]
Originally asked (and answered) [on twitter](https://twitter.com/Esclaponr/status/1356995385525325827):

How can we use tidy.js in rmarkdown?

**Comment by gadenbuie** [comment]
For anyone interested in using tidy.js with R Markdown (or Shiny apps), I've put together an R package that provides d3-array and the UMD version of tidy.js: https://github.com/gadenbuie/tidyjs-r

**Comment by pbeshai** [comment, marked as answer]
*switching to answer format*

[As shown on twitter](https://twitter.com/Esclaponr/status/1356995385525325827), this can be done this way or via @gadenbuie's new tidyjs-r package.

````rmarkdown
---
title: "Untitled"
output: html_document
---

<script src="https://d3js.org/d3-array.v2.min.js"></script>
<script src="https://www.unpkg.com/@tidyjs/tidy/dist/umd/tidy.min.js"></script>

```{js}
// everything is available under the Tidy namespace
const { tidy, mutate, arrange, desc } = Tidy;

const data = [
  { a: 1, b: 10 }, 
  { a: 3, b: 12 }, 
  { a: 2, b: 10 }
]

const results = tidy(
  data, 
  mutate({ ab: d => d.a * d.b }),
  arrange(desc('ab'))
)
```
````
